src/Entities/RulesEngine/Condition.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * This source file is proprietary and part of Rebilly.
+ *
+ * (c) Rebilly SRL
+ *     Rebilly Ltd.
+ *     Rebilly Inc.
+ *
+ * @see https://www.rebilly.com
+ */
+
+namespace Rebilly\Entities\RulesEngine;
+
+use DomainException;
+use Rebilly\Rest\Resource;
+
+/**
+ * Class Condition.
+ */
+class Condition extends Resource
+{
+    public const UNEXPECTED_OP = 'Unexpected operation. Only %s operations are supported';
+
+    public const OP_AND = 'and';
+
+    public const OP_OR = 'or';
+
+    public const OP_NOT = 'not';
+
+    public const OP_BETWEEN = 'between';
+
+    public const OP_EQUALS = 'equals';
+
+    public const OP_IN = 'in';
+
+    public const OP_GREATER_THAN = 'gt';
+
+    public const OP_GREATER_THAN_OR_EQUAL = 'gte';
+
+    public const OP_LESS_THAN = 'lt';
+
+    public const OP_LESS_THAN_OR_EQUAL = 'lte';
+
+    /**
+     * @return string[]|array
+     */
+    public static function ops(): array
+    {
+        return [
+            self::OP_AND,
+            self::OP_OR,
+            self::OP_NOT,
+            self::OP_BETWEEN,
+            self::OP_EQUALS,
+            self::OP_IN,
+            self::OP_GREATER_THAN,
+            self::OP_GREATER_THAN_OR_EQUAL,
+            self::OP_LESS_THAN,
+            self::OP_LESS_THAN_OR_EQUAL,
+        ];
+    }
+
+    /**
+     * @param array $data
+     *
+     * @return Condition
+     */
+    public static function createFromData(array $data): self
+    {
+        return new self($data);
+    }
+
+    /**
+     * @return string
+     */
+    public function getOp(): string
+    {
+        return $this->getAttribute('op');
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return $this
+     */
+    public function setOp($value): self
+    {
+        if (!in_array($value, self::ops(), true)) {
+            throw new DomainException(sprintf(self::UNEXPECTED_OP, implode(', ', self::ops())));
+        }
+
+        return $this->setAttribute('op', $value);
+    }
+
+    /**
+     * @return Condition
+     */
+    public function getCondition(): ?self
```

Comment:
in setter we define that `Condition` or `array` can be set, in getter return type is `Condition` or null and in PHPDoc return type is only `Condition`, where is the truth? :)